core/Tracker/GoalManager.php

```diff
@@ -877,10 +877,22 @@ private function triggerHookOnDimensions(Request $request, $dimensions, $hook, $
 
     private function getGoalFromVisitor(VisitProperties $visitProperties, Request $request, $action)
     {
+        $lastVisitTime = $visitProperties->getProperty('visit_last_action_time');
```

Comment:
this part is not 100% related to this PR. It refs https://github.com/matomo-org/wp-matomo/issues/215#issuecomment-603514307 figured since it is related to visit last action time etc might fix that as well.